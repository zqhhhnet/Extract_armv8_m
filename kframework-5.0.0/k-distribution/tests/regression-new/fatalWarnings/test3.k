module TEST3-SYNTAX

endmodule

module TEST3
  imports TEST3-SYNTAX
  imports BOOL

  syntax Bool ::= "foo" [function]

  rule 0 => 1
    requires (foo andBool (foo orBool foo) orBool foo)
endmodule
