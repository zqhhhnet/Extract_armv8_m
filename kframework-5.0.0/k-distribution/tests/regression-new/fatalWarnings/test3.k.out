[Error] Compiler: Had 1 parsing errors.
[Error] Inner Parser: Parsing ambiguity. Arbitrarily choosing the first.
1: syntax Bool ::= Bool "andBool" Bool [boolOperation() function() functional()
hook(BOOL.and) klabel(_andBool_) latex({#1}\wedge_{\scriptstyle\it Bool}{#2})
left() smt-hook(and) symbol()]
   
`_andBool_`(`foo_TEST3`(.KList),`_orBool_`(`_orBool_`(`foo_TEST3`(.KList),`foo_TEST3`(.KList)),`foo_TEST3`(.KList)))
2: syntax Bool ::= Bool "orBool" Bool [boolOperation() function() functional()
hook(BOOL.or) klabel(_orBool_) latex({#1}\vee_{\scriptstyle\it Bool}{#2})
left() smt-hook(or)]
   
`_orBool_`(`_andBool_`(`foo_TEST3`(.KList),`_orBool_`(`foo_TEST3`(.KList),`foo_TEST3`(.KList))),`foo_TEST3`(.KList))
	Source(test3.k)
	Location(12,15,12,54)
